twisted-infra/braid
# [#81] Use upstream buildbot version

Scope
=====

This tries to fix #81 by using an upstream version of buildbot.


Changes
=======

buildbot eight branch (unreleased) branch is used, as 0.8.12 is buggy.
Also, the branch is still hosted in the braid/buildbot fork so that we can control its updates.

PB listened was removed as I don't think that it is used.

Some changes from the webstatus templates were reverted so that the diff is minimal. This should help with future upgrades. So the diff here for templates is big but comparing with upstream the changes are small and isolated

The buildslave configuration was updated to also install sqlite-dev so that we can run the tests.

The addStep warning are there, but I will work in a separate branch to get rid of them.

While working on this I found a bug with the `countFailedTests` helper as it was crashing if the output has no test result... for example when the test failed before running trial

How to test
=========

I have tested this on my vagrant.

This PR is on top of #204 as there I have fixed the braid state for the vagrant VM. So first go and review and merge #204 :)

Here are the commands I used
```
fab config.vagrant buildbot.stop
fab config.vagrant buildbot.update
fab config.vagrant buildbot.start
fab config.vagrant buildslave.stop
fab config.vagrant buildslave.update
fab config.vagrant buildslave.start
```

After that you can go at http://172.16.255.140:8080/ and trigger a few tests

I have tested it on linux.

I think that we should deploy it and then see if we get any errors or regression.
